x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/execution/search/extractor/AbstractFieldHitExtractor.java

```diff
@@ -33,15 +33,24 @@ public abstract class AbstractFieldHitExtractor implements HitExtractor {
     private final ZoneId zoneId;
     private final boolean arrayLeniency;
 
+    protected boolean allowMultiValue = false;
```

Comment:
Be careful with this new parameter, from serialization point of view and BWC checks...